src/logic/taskcluster.ts

```diff
@@ -0,0 +1,43 @@
+// This file contains logic for the Taskcluster Third-Party Login
+
+import { prodFirefoxRootUrl, getRootUrl } from '../common/constants';
+import { UserCredentials } from '../types/types';
+import { generateNonce } from '../utils/helpers';
+
+// For testing purposes
+const treeherderClientId = `treeherder-localhost-5000-client`;
+const treeherderRedirectURI = `http://localhost:5000/taskcluster-auth`;
+
+const getAuthCode = (defaultRootUrl: string) => {
+  const nonce = generateNonce();
+  localStorage.setItem('requestState', nonce);
```

Comment:
Probably better to use `sessionStorage` (marginally better).
Can you add a small comment explaining why we need to store it (cf my other comment above)